src/io/hdf5_writer.cpp

```diff
@@ -23,7 +23,15 @@ HDF5Writer::HDF5Writer(const std::string& report_name)
     std::tie(collective_list_, independent_list_) = Implementation::prepare_write(report_name,
                                                                                   plist_id);
     // Create hdf5 file named after the report_name
-    const std::string file_name = report_name + ".h5";
+    std::string file_name = report_name + ".h5";
+
+#if defined(HAVE_MPI)
```

Comment:
where this `HAVE_MPI` comes from?

Reply:
This is already defined and utilized in other places, to distinguish between the serial HDF5 and the parallel HDF5. See for instance:

https://github.com/BlueBrain/libsonatareport/blob/master/src/library/implementation_interface.hpp#L13

Reply:
Ok. 
(nitpick for a separate enhancement / PR : ideally this should be `SONATA_REPORT_HAVE_MPI`)

Reply:
I'd say why not use `H5_HAVE_PARALLEL` and avoid any sketchy definitions at all?

Reply:
Ok, I have updated it, I believe it is cleaner right now. Also, I checked with @jorblancoa to ensure all the changes were correct, specially in regards to the integration tests.